pkg/s3io/iosource.go

```diff
@@ -0,0 +1,24 @@
+package s3io
+
+import (
+	"errors"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/brimsec/zq/pkg/iosource"
+)
+
+var DefaultSource = &Source{}
+var _ iosource.Source = DefaultSource
+
+type Source struct {
+	Config *aws.Config
+}
+
+func (l *Source) NewWriter(path string) (io.WriteCloser, error) {
+	return NewWriter(path, l.Config)
+}
+
+func (l *Source) NewReader(path string) (io.ReadCloser, error) {
```

Comment:
Nit: `l` is an odd name for the receiver.
```suggestion
func (s *Source) NewWriter(path string) (io.WriteCloser, error) {
	return NewWriter(path, s.Config)
}

func (s *Source) NewReader(path string) (io.ReadCloser, error) {
```